lib/core/test/integration/Test/Integration/Scenario/CLI/Wallets.hs

```diff
@@ -142,7 +164,52 @@ spec = do
         err `shouldBe` "Ok.\n"
         out `shouldNotContain` "CLI Wallet"
         c `shouldBe` ExitSuccess
-
   where
     emptyWallet' :: Context t -> IO String
     emptyWallet' = fmap (T.unpack . view walletId) . emptyWallet
+
+specPorts :: Int -> SpecWith (Context t)
+specPorts defaultPort = do
+    it "PORTS_01 - Can create wallet with specifying port" $ \_ -> do
+        let walletName = "Wallet created via CLI"
+        checkSimpleCreateWallet
+            (Just (show defaultPort))
+            "15"
+            walletName
+            (`shouldBe` ExitSuccess)
+            (`shouldContain` "Ok.\n")
+            (`shouldContain` walletName)
+    xit "PORTS_02 - Can create wallet with relying on default port" $ \_ -> do
+        let walletName = "Wallet created via CLI"
+        checkSimpleCreateWallet
+            Nothing
+            "15"
+            walletName
+            (`shouldBe` ExitSuccess)
+            (`shouldContain` "Ok.\n")
+            (`shouldContain` walletName)
+    it "PORTS_03 - Cannot create wallet with wrong port" $ \_ -> do
+        let walletName = "Wallet created via CLI"
+        checkSimpleCreateWallet
+            (Just (show (defaultPort + 1)))
+            "15"
+            walletName
+            (`shouldBe` ExitSuccess)
+            (`shouldContain` "does not exist (Connection refused))\n")
+            (`shouldNotContain` walletName)
+
+checkSimpleCreateWallet
+    :: Maybe String
+    -> String
+    -> String
+    -> (ExitCode -> IO ())
+    -> (String -> IO ())
+    -> (String -> IO ())
```

Comment:
not found of this helper, it's really "stringly-typed" and is more cryptic than readable IMO. I'd rather inline it in all the test above as it reads better inlined.